### PR TITLE
diff changes

### PR DESCRIPTION
**Pull request by schwiti6190**
- improves loading/unloading with multiple fillunits in mode 1/4/8
- slightly improves cover handling
- improves triggerspeed near stationary triggers

### CODE REVIEW

File: AIDriver.lua [hunk 1]
```diff
@@ -493,6 +500,49 @@ function AIDriver:driveCourse(dt)
...
+	
+	return (self.currentlyInTrigger or self.triggerHandler:isInTrigger()) and self.bunkerSiloManager == nil or false
+end
+
+function AIDriver:setIsInTrigger()
+	self.currentlyInTrigger = true
+end
+
+
+
```
Comment (schwiti6190) [inline AIDriver.lua:519]: Course:hasTriggerRaycast(startIx,driverClass,driverCallback)  callback,
sadly it's not working as intended, as raycast only seem to detect node objects, but not a trigger bounding box of for example a fueltrigger of a fuel refill trailer.
=> so it's not used right now

File: FieldSupplyAIDriver.lua [hunk 2]
```diff
@@ -21,90 +21,127 @@ Field Supply AI Driver to let fill tools with digestate or liquid manure on the
 Also known as mode 8
 ]]
 
---TODO: Have FieldSupplyAIDriver be derived from GrainTransportDriver and not FillableFieldworkAIDriver,
---		as there is no need for FieldworkAIDriver functions 
-
----@class FieldSupplyAIDriver : FillableFieldworkAIDriver
-FieldSupplyAIDriver = CpObject(FillableFieldworkAIDriver)
+---@class FieldSupplyAIDriver : GrainTransportAIDriver
+FieldSupplyAIDriver = CpObject(GrainTransportAIDriver)
```
Comment (schwiti6190) [inline FieldSupplyAIDriver.lua:26]: Changes the inheritance to GrainTransportAIDriver, as we don't need the field course and so on. 
Might have to heck if this change impactes autodrive setup.

File: AIDriver.lua [hunk 3]
```diff
@@ -493,6 +500,49 @@ function AIDriver:driveCourse(dt)
 	end
 end
```
Comment (schwiti6190) [inline AIDriver.lua:502]: for now we are still using the old self.triggerHandler:isInTrigger(), which uses the old triggers.lua code, might be a good idea to not depent on that code in the long run

**Review by pvaiko — CHANGES_REQUESTED** [review]
Looks like I forgot to submit this on Friday...

**Comment by schwiti6190** [comment]
@pvaiko thx for your feedback, I will have a look and try to adjust your ideas :)

**Pull request closed without merging**